packages/components/src/internal/components/editable/EditableGridPanelForUpdate.tsx

```diff
@@ -89,14 +72,14 @@ export class EditableGridPanelForUpdate extends React.Component<Props, State> {
                 data,
                 index
             );
-        });
-        this.props.setIsDirty?.(true);
-    };
-
-    onSubmit = (): void => {
-        const { onComplete, updateRows, idField, selectionData, singularNoun } = this.props;
-        const { dataModels, editorModels } = this.state;
-
+            setDataModels(modelUpdates.dataModels);
+            setEditorModels(modelUpdates.editorModels);
+            setIsDirty?.(true);
```

Comment:
Something you may want to consider adding here is a check for if the "data changed" when setting dirty by calling `modelUpdates.editorModels,`. This is a somewhat new fangled notion that I introduced for editable grids which is to differentiate events on the modals into categories/types so we can differentiate behaviors. For example, just clicking around on the editable grid without actually causing a data change will not result in a dirty state.

```tsx
if (modelUpdates.editorModels.find(model => model.isDataChangeEvent(event))) {
    setIsDirty?.(true);
}
```